src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+public class Ball {
+
+    private String color;
+    private int number;
+
+    public Ball(int number, String color) {
+
+        boolean checkColor = isColor(color);
+        if (number < 100 && number > 0 && checkColor) {
+            this.number = number;
+            this.color = color;
+        } else {
+            System.out.println("Wrong color or Number is not between 0 and 100 ");
+        }
+    }
+
+    public void setNumber(int number) {
+
+        if (number < 100 && number > 0) {
+            this.number = number;
+        } else {
+            System.out.println("Ball number has to between 0 and 100!");
+        }
+    }
+
+    public void setColor(String color) {
+
+        if (isColor(color)) {
+            this.color = color;
+        } else {
+            System.out.println("Wrong color");
+        }
+    }
+
+    private boolean isColor(String color) {
+
+        boolean checkColor = false;
+        for (int i = 0; i < Color.values().length; i++) {
+            if (color.equals(Color.values()[i].toString())) {
+                checkColor = true;
+            }
+        }
+        return checkColor;
+    }

```

Comment:
ball class shouldn't be bothered with logic of color supplier (like knowing limits for number 0 to 100), This class should only make it possible to create a ball instance given some name, and some color
```suggestion
    public Ball(int number, String color) {
         this.number = number;
         this.color = color;
     }   

```